packages/esm-patient-conditions-app/src/conditions/conditions-widget.component.tsx

```diff
@@ -24,83 +24,165 @@ import {
   useLayoutType,
   useSession,
 } from '@openmrs/esm-framework';
-import { createPatientCondition, CodedCondition, useConditionsSearch } from './conditions.resource';
+import {
+  createPatientCondition,
+  CodedCondition,
+  useConditionsSearch,
+  ApiCondition,
+  editPatientCondition,
+} from './conditions.resource';
 import styles from './conditions-form.scss';
 
 interface ConditionsWidgetProps {
   patientUuid: string;
   closeWorkspace?: () => void;
   setHasSubmissibleValue?: (value: boolean) => void;
   submissionNotifier: BehaviorSubject<{ isSubmitting: boolean }>;
+  condition?: ApiCondition;
+  context?: string;
 }
 
+const getFieldValue = (cells, fieldName) => cells?.find((c) => c?.info?.header === fieldName)?.value;
+
 const ConditionsWidget: React.FC<ConditionsWidgetProps> = ({
   patientUuid,
   closeWorkspace,
   setHasSubmissibleValue,
   submissionNotifier,
+  condition,
+  context,
 }) => {
   const { t } = useTranslation();
   const session = useSession();
   const { mutate } = useSWRConfig();
   const isTablet = useLayoutType() === 'tablet';
-  const [clinicalStatus, setClinicalStatus] = useState('active');
+
+  const [clinicalStatus, setClinicalStatus] = useState(
+    context === 'editing' ? getFieldValue(condition?.cells, 'clinicalStatus').toLowerCase() : 'active',
+  );
   const [endDate, setEndDate] = useState(null);
-  const [onsetDate, setOnsetDate] = useState(new Date());
-  const [conditionToLookup, setConditionToLookup] = useState<null | string>('');
-  const [selectedCondition, setSelectedCondition] = useState(null);
+  const [onsetDate, setOnsetDate] = useState(
+    context === 'editing' ? new Date(getFieldValue(condition?.cells, 'onsetDateTime')) : new Date(),
+  );
+  const [conditionToLookup, setConditionToLookup] = useState<null | string>(
+    context === 'editing' ? getFieldValue(condition?.cells, 'display') : '',
+  );
 
   const { conditions, isSearchingConditions } = useConditionsSearch(conditionToLookup);
 
+  const [selectedCondition, setSelectedCondition] = useState(null);
+
+  // Populate the selected condition when in edit mode
+  useEffect(() => {
```

Comment:
I'm not too stoked about all these `useEffect`s. However, I haven't found a way to get rid of them that doesn't completely bork the edit feature. Something worth thinking about even beyond this PR.

Reply:
You are right, I am also not a fun of `useEffects` but for this case since the conditions delay a bit, it was hard to populate the selected condition, so a useEffect was the best option.